nim/tests/test_routes.py

```diff
@@ -0,0 +1,131 @@
+import json
+
+import pytest
+
+from nim.app import app
+from nim.game import (
+    NUM_LIMIT,
+    DEFAULT_MIN,
+    DEFAULT_MAX,
+    DEFAULT_PILES,
+)
+
+
+@pytest.fixture
+def client_app():
+    return app.test_client()
+
+
+def assert_new_game_success(client, request):
+    """
+    Asserts post to /new gets 200 & state is not empty. Checks response state
+    is in line with request. Uses arguments passed or DEFAULT
+    :param client: test_client
+    :param request: Desired game conditions
+    """
+    response = client.post(
+        '/new',
+        content_type='application/json',
+        data=json.dumps(request)
+    )
+    assert response.status_code == 200
+    response = json.loads(response.data)
+    assert 'state' in response
+    state = response['state']
+    assert all(isinstance(x, int) for x in state)
+    # Checking num piles returned is requested or default
+    assert request.get('piles', DEFAULT_PILES) == len(state)
+    # Checking piles returned fall within request or default
+    assert ((request.get('min', DEFAULT_MIN) <= min(state) <= max(state)
+            <= request.get('max', DEFAULT_MAX)) <= NUM_LIMIT)
```

Comment:
You have extra parentheses in there.
Also, Python will convert this to:
```
    assert (request.get('min', DEFAULT_MIN) <= min(state) and
            min(state) <= max(state) and
            max(state) <= request.get('max', DEFAULT_MAX) and
            request.get('max', DEFAULT_MAX) <= NUM_LIMIT)
```
The 2nd test is useless (of course `min(state) <= max(state)`).
May be cleaner to split the tests and also define a new global variable (we really need to find better variable names, but don't worry about it for now):
```
    # Check that the min value was respected
    assert NUM_LIMIT_MIN <= request.get('min', DEFAULT_MIN) <= min(state)
    # Check that the max value was respected
    assert max(state) <= request.get('max', DEFAULT_MAX) <= NUM_LIMIT_MAX
```
We should also have another test function that does something like:
```
def test_constants():
    assert all(NUM_LIMIT_MIN <= x <= NUM_LIMIT_MAX
               for x in (DEFAULT_MIN, DEFAULT_MAX, DEFAULT_PILES))
```